post-processor/vsphere/post-processor.go

```diff
@@ -115,6 +115,33 @@ func (p *PostProcessor) Configure(raws ...interface{}) error {
 	return nil
 }
 
+func (p *PostProcessor) generateURI() (*url.URL, error) {
+	// use net/url lib to encode and escape url elements
+	ovftool_uri := fmt.Sprintf("vi://%s/%s/host/%s",
+		p.config.Host,
+		p.config.Datacenter,
+		p.config.Cluster)
+
+	if p.config.ResourcePool != "" {
+		ovftool_uri += "/Resources/" + p.config.ResourcePool
+	}
+
+	u, err := url.Parse(ovftool_uri)
+	if err != nil {
+		return nil, fmt.Errorf("Couldn't generate uri for ovftool: %s", err)
+	}
+	u.User = url.UserPassword(p.config.Username, p.config.Password)
+
+	if p.config.ESXiHost != "" {
+		if ipv4Regex.MatchString(p.config.ESXiHost) {
+			u.RawQuery = "ip=" + url.QueryEscape(p.config.ESXiHost)
+		} else if hostnameRegex.MatchString(p.config.ESXiHost) {
+			u.RawQuery = "dns=" + url.QueryEscape(p.config.ESXiHost)
+		}
+	}
```

Comment:
This might lose potentially already present query parameters, one thing we can do:
```suggestion
	if p.config.ESXiHost != "" {
		q := u.Query()
		if ipv4Regex.MatchString(p.config.ESXiHost) {
			q.Add("ip", p.config.ESXiHost)
		else if hostnameRegex.MatchString(p.config.ESXiHost) {
			q.Add("dns", p.config.ESXiHost)
		}
		u.RawQuery = q.Encode()
	}
```